bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,875 @@
+                break;
+            } else if (bypass != null) {
+                if ((currentState.isPowerlinkMode() == null) || currentState.isPowerlinkMode().equals(Boolean.FALSE)) {
+                    logger.warn("PowerMax alarm binding: Bypass option only supported in Powerlink mode");
+                } else if (!PowerMaxPanelSettings.getThePanelSettings().isBypassEnabled()) {
+                    logger.warn("PowerMax alarm binding: Bypass option not enabled in panel settings");
+                } else {
+                    comm.sendZoneBypass(bypass.booleanValue(), zoneNr,
+                            PowerMaxPanelSettings.getThePanelSettings().getFirstPinCode());
+                }
+                break;
+            } else if (commandStr != null) {
+                if (commandStr.equalsIgnoreCase("get_event_log")) {
+                    comm.requestEventLog(currentState.isPowerlinkMode()
+                            ? PowerMaxPanelSettings.getThePanelSettings().getFirstPinCode() : pinCode);
+                } else if (commandStr.equalsIgnoreCase("download_setup")) {
+                    if ((currentState.isPowerlinkMode() == null)
+                            || currentState.isPowerlinkMode().equals(Boolean.FALSE)) {
+                        logger.warn("PowerMax alarm binding: download setup only supported in Powerlink mode");
+                    } else {
+                        comm.startDownload();
+                        if (currentState.getLastKeepAlive() != null) {
+                            currentState.setLastKeepAlive(System.currentTimeMillis());
+                        }
+                    }
+                } else if (commandStr.equalsIgnoreCase("log_setup")) {
+                    settings.log();
+                } else if (commandStr.equalsIgnoreCase("help_items")) {
+                    settings.helpItems();
+                } else {
+                    logger.warn("PowerMax alarm binding: rejected command {}", commandStr);
+                }
+                break;
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+        logger.debug("updated(): updating configuration");
+
+        closeConnection();
+
+        serialPort = null;
+        ipAddress = null;
+        tcpPort = 0;
+        motionOffDelay = DEFAULT_MOTION_OFF_DELAY;
+        allowArming = false;
+        allowDisarming = false;
+        forceStandardMode = false;
+        panelType = PowerMaxPanelType.POWERMAX_PRO;
+        autoSyncTime = false;
+        pinCode = null;
+
+        PowerMaxReceiveType.POWERLINK.setHandlerClass(PowerMaxPowerlinkMessage.class);
+
+        if (config == null) {
+            logger.warn("PowerMax alarm binding: missing configuration");
+            this.setProperlyConfigured(false);
+            throw new ConfigurationException(null, "missing configuration");
+        }
```

Comment:
I thought the proper response to receiving a `null` config in this method was to do nothing at all in this method?  Could you verify?